### ISSUE TITLE
Option for providing client TLS certificate

### ISSUE THREAD

**Issue by natlibfi-arlehiko** [indexdata/yaz#40]
Please implement an option to provide the client's TSL certificate. Currently you need a proxy server like stunnel/haproxy to provide the certificate if the server verifies it.

**Comment by MikeTaylor** [comment]
This is by design. YAZ is a Z39.50 toolkit, not a web server. We could keep adding features and features to it, until it contains an email reader and (in accordance with Greenspun's tenth rule), an ad-hoc, informally-specified, bug-ridden, slow implementation of half of Common Lisp. But we prefer the Unix philosophy of keeping each tool small and clean, doing one thing well. haproxy is a fine tool for supporting TLS.

**Comment by natlibfi-arlehiko** [comment]
But the support is there, since `yaz-client` allows you to connect to TLS servers (By prefixing the URL with `ssl:`) but it apparently uses a static TLS cert.

**Comment by MikeTaylor** [comment]
Thanks for spotting that. I have re-opened the issue, and will leave to to @adamdickmeiss to comment.

**Comment by adamdickmeiss** [comment]
Yes. Seems like a good idea.. Like -E option for curl and --certificate option for wget.

**Comment by natlibfi-arlehiko** [comment]
So is this going to be implemented only in the CLI or in the library as well?

**Comment by adamdickmeiss** [comment]
Library ncluding the ZOOM API.

**Comment by adamdickmeiss** [comment]
https://github.com/indexdata/yaz/pull/192
